### ISSUE TITLE
The "embodied" properties need to have their ranges changed

### ISSUE THREAD

**Issue by stuartasutton** [CredentialEngine/Schema-Development#812]
The [ceasn:skillEmbodied](https://purl.org/ctdlasn/terms/skillEmbodied), [ceasn:knowledgeEmbodied](https://purl.org/ctdlasn/terms/knowledgeEmbodied), [ceasn:abilityEmbodied](https://purl.org/ctdlasn/terms/abilityEmbodied) and [ceasn:taskEmbodied](https://purl.org/ctdlasn/terms/taskEmbodied) properties have a range of `xsd:anyURI`. They all should have a range of [ceasn:Competency](https://purl.org/ctdlasn/terms/Competency) since, with the possible exception of task reference KSAs which for us are competencies.

**Comment by siuc-nate** [comment]
I think we made these URIs deliberately so we could reference KSAs from other sources like EMSI and tasks from other sources like NIST.

**Comment by stuartasutton** [comment]
Yes, but the things being "embodied are KSAs etc. not just any URI.

**Comment by philbarker** [comment]
1. Stuart is right, the purpose of declaring the range of a property is to say what type of thing you will get data for if you resolve it.
2. we probably shouldn't have xsd:anyURI as the range for any property as it is a datatype for literals not a class. Even if it were a class it would be the class of URIs so having it in the range means the property points to a URI and you will get a decsription of that URI.

**Comment by siuc-nate** [comment]
I would push back on changing xsd:anyURI across the board, as our systems (including CE, CaSS, and the Registry) are all using our web.config and thus relying on various things being xsd:anyURI.  Unless we prepare in advance for a change over to rdfs:Resource, changing our context file like that will probably break things in ways that may or may not be easy to detect.

**Comment by stuartasutton** [comment]
Let's not get too sidetracked with the xsd:anyURI (separate issue) but keep focused here on the distinction between an occupation as a distinct class (e.g., ceterm:Occupation)  and as a concept in a classification scheme.

**Comment by philbarker** [comment]
Yes, but let's not get sidetracked here.

**Comment by siuc-nate** [comment]
@stuartasutton @philbarker is this something that needs more discussion? Otherwise it seems fairly straightforward.

**Comment by stuartasutton** [comment]
If the conclusion is to change the range on these embodied properties to ceasn:Competency, then we are done.

**Comment by siuc-nate** [comment]
Proposal:

**Remove:**
> **Subject:** ceasn:skillEmbodied, ceasn:knowledgeEmbodied, ceasn:abilityEmbodied, ceasn:taskEmbodied
> **Predicate:** schema:rangeIncludes
> **Object:** xsd:anyURI

**Add:**
> **Subject:** ceasn:skillEmbodied, ceasn:knowledgeEmbodied, ceasn:abilityEmbodied
> **Predicate:** schema:rangeIncludes
> **Object:** ceasn:Competency

> **Subject:** ceasn:taskEmbodied
> **Predicate:** schema:rangeIncludes
> **Object:** ceterms:Task

@jeannekitchens This does represent a potentially breaking change for CaSS, so we will need to work with them to prepare before we actually make this change

**Comment by stuartasutton** [comment]
I think there needs to be a modification to the proposed ceterms:taskEmbodied adding ceterms:Competency since there can be competency framework descriptions with competency nodes identified as "task":

> **Subject:** ceasn:taskEmbodied
> **Predicate:** schema:rangeIncludes
> **Object:** ceterms:Competency, ceterms:Task

**Comment by siuc-nate** [comment]
Yes, we would want to avoid breaking things.  There is also the matter of potentially breaking other consuming systems at this point to consider as well.

For what it's worth, xsd:anyURI can already be used to point at Competencies/Tasks (among other things), so this isn't a terribly necessary change to make - I believe the Finder, for example, already checks to see if the URI provided is a registry URI or returns JSON-LD and handles that accordingly; otherwise it just gets treated like a webpage link.

In short: While this would make the schema a little cleaner, I think we are okay leaving it alone.

**Comment by siuc-nate** [comment]
This has resurfaced again in https://github.com/CredentialEngine/CredentialRegistry/issues/701#issuecomment-2107771997

If we can ensure this doesn't break CaSS, then this should probably move forward.

@jeannekitchens @philbarker @Lomilar @mparsons-ce thoughts?

**Comment by philbarker** [comment]
Looks like they should have a rangeIncludes of [ceasn:Competency](https://purl.org/ctdlasn/terms/Competency)

**Comment by siuc-nate** [comment]
@jeannekitchens @philbarker This one was flagged back in 2022 as "ready to implement", but implementing it would break CaSS.

Thoughts?

**Comment by philbarker** [comment]
The solution that is ready to implement is right for CTDL, I have no opinion on the consequences for CaSS.
